cmd/ddev/cmd/config.go

```diff
@@ -33,22 +39,43 @@ var ConfigCommand = &cobra.Command{
 
 		c, err := ddevapp.NewConfig(appRoot, provider)
 		if err != nil {
-			// If there is an error reading the config and the file exists, we're not sure
-			// how to proceed.
-			if c.ConfigExists() {
-				util.Failed("Could not read config: %v", err)
-			}
+			util.Failed("Could not create new config: %v", err)
 		}
 
 		// Set the provider value after load so we can ensure we use the passed in provider value
 		// for this configuration.
 		c.Provider = provider
+		c.Name = siteName
+		c.Docroot = docrootRelPath
 
-		err = c.Config()
-		if err != nil {
-			util.Failed("There was a problem configuring your application: %v\n", err)
-		}
+		if siteName == "" && docrootRelPath == "" && pantheonEnvironment == "" {
```

Comment:
I have mixed feelings about this approach. I'd really like to see a way for providers to each have their own top-level command and register the flags they need. It doesn't feel like this method scales at all beyond our current single provider. it also consolidates a lot of untested logic into our cobra commands, which is a method we've mostly been moving away from. 

On the other hand, I realize this is a blocking issue that we're trying to move on relatively quickly. I also realize we only have a single provider right now, so over-engineering for the future has its own risk. Especially when we have no immediate plans to work on additional providers.

Reply:
I think you've nailed it. You're definitely right that this doesn't scale very well, but that there's no reason to deal with the architecture until the next provider.